File: trabalho/Solucao/Exercicio_2/Banco/src/main/java/banco/dominio/Conta.java
package banco.dominio;

import banco.dominio.excecoes.ExcecaoChequeEspecial;

public class Conta {

    protected double saldo;

    public Conta(double saldoInicial) {
        this.saldo = saldoInicial;
    }

    public double getSaldo() {
        return saldo;
    }

    public boolean depositar(double valor) {
        saldo = saldo + valor;
        return true;
    }

    public void sacar(double valor) throws ExcecaoChequeEspecial {
        if ((this.saldo - valor) >= 0) {
            saldo = saldo - valor;
         } else {
            double deficit = valor - this.saldo;
            throw new ExcecaoChequeEspecial("Saldo insuficiente", deficit);
        }
     }
}

File: javascript/Conta.js
module.exports = class Conta {
  constructor(saldo) {
    this.saldo = saldo;
  }
  getSaldo() {
    return this.saldo;
  }
  depositar(valor) {
    this.saldo = this.saldo + valor;
    return true;
  }
  sacar(valor) {
    if (this.saldo - valor >= 0) {
      this.saldo = this.saldo - valor;
    } else {
      throw new ExcecaoChequeEspecial("saldo insuficiente", valor - this.saldo);
    }
  }
};
